### PR TITLE
ROC-5841 Map DQ hearing court to LA UI

### PR DESCRIPTION
**Pull request by simad**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5841

### Change description ###

Mapping the hearing court details from DQ to the LA UI. Also fixing a bug in which case details were lost after drawing the order 😬 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/GenerateOrderCallbackHandlerTest.java [hunk 1]
```diff
@@ -80,10 +90,23 @@ public void setUp() {
...
-            docAssemblyTemplateBodyMapper
-        );
+            docAssemblyTemplateBodyMapper,
+            caseDetailsConverter);
         ReflectionTestUtils.setField(generateOrderCallbackHandler, "templateId", "testTemplateId");
         ccdCase = SampleData.getCCDCitizenCase(Collections.emptyList());
+        Claim claim =
+            SampleClaim.builder()
+                .withResponse(
+                    FullDefenceResponse.builder()
```
Comment (karthikkannankainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/GenerateOrderCallbackHandlerTest.java]: SampleResponse may have this FullDefenceResponse provider. If not pls add it to there so others can use too. Thanks

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/ResourceLoader.java [hunk 2]
```diff
@@ -43,6 +43,11 @@ public static CaseDetails successfulCoreCaseDataStoreSubmitResponse() {
         return jsonMapper.fromJson(response, CaseDetails.class);
     }
 
+    public static CaseDetails successfulCoreCaseDataStoreSubmitResponseWithDQ() {
+        String response = getResource("/core-case-data/submit-response.success-with-dq.json");
```
Comment (karthikkannankainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/utils/ResourceLoader.java]: Curiosity - Why `.` between words in file name?

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/util/SampleData.java [hunk 3]
```diff
@@ -339,34 +340,36 @@ public static CCDApplicant getCCDApplicantIndividual() {
             .representativeOrganisationDxAddress("dx123")
             .build();
     }
-
-    public static CCDApplicant getIndividualApplicantWithDQ() {
+    
+    public static CCDRespondent getIndividualRespondentWithDQ() {
```
[the remark this answers is not in the document]
Comment (simad) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/util/SampleData.java]: 👍 will do

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/GenerateOrderCallbackHandler.java [hunk 4]
```diff
@@ -160,25 +172,37 @@ private CallbackResponse generateOrder(CallbackParams callbackParams) {
             .build();
     }
 
-    private void addCourtData(CCDCase ccdCase, Map<String, Object> data) {
+    private void addCourtData(Claim claim, Map<String, Object> caseData, Map<String, Object> data) {
+        CCDCase ccdCase = jsonMapper.fromMap(caseData, CCDCase.class);
+        CCDRespondent respondent = ccdCase.getRespondents().get(0).getValue();
+
+        CCDResponseRejection claimantResponse = Optional.ofNullable(respondent.getClaimantResponse())
+            .map(r -> (CCDResponseRejection) r)
```
[the remark this answers is not in the document]
Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/GenerateOrderCallbackHandler.java]: 👍

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GenerateOrderCallbackHandlerTest.java [hunk 5]
```diff
@@ -96,7 +101,7 @@ public void shouldPrepopulateFieldsOnAboutToStartEvent() throws Exception {
             .containsExactlyInAnyOrder("DOCUMENTS", "EYEWITNESS");
         assertThat(responseData.get("docUploadForParty")).isEqualTo("BOTH");
         assertThat(responseData.get("eyewitnessUploadForParty")).isEqualTo("BOTH");
-        assertThat(responseData.get("preferredCourt")).isEqualTo("Preferred court");
+        assertThat(responseData.get("preferredDQCourt")).isEqualTo("Preferred court");
```
[the remark this answers is not in the document]
Comment (simad) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GenerateOrderCallbackHandlerTest.java]: I usually don't mind having plain strings in tests, will take a look :)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/ResourceLoader.java (hunk 2 above)
Comment (simad) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/utils/ResourceLoader.java]: was just following the existing patter 🤷‍♀

**Pull request merged**